### ISSUE TITLE
Register JerseyViolationExceptionMapper only if not overriden in ResourceTestRule

### ISSUE THREAD

**Issue by lingz** [dropwizard/dropwizard#1381]
Right now inside ResourceTestResourceConfig inside ResourceTestRule, it is hardcoded to always register JerseyViolationException. This breaks in cases where we have our own ExceptionMapper for JerseyViolationException, and want to test the response from that as the default exception mapper will always have priority.

A better behaviour is to check if any exception mappers already exist for JerseyViolationException and only add the ExceptionMapper if it does not exist already.

The line is found here:
https://github.com/dropwizard/dropwizard/blob/master/dropwizard-testing/src/main/java/io/dropwizard/testing/junit/ResourceTestRule.java#L158

**Comment by nickbabcock** [comment]
Thanks for trying out 1.0! It looks like this problem arose because of my misunderstanding that registering exception mappers non-deterministic which one is used. There was a pull request to revert registering exception mappers (#1148) in a test resource, but it looks like `JerseyViolationExceptionMapper` is a hold out.

Other related commits and pull requests that show how this non-determinism really threw me off :blush: 

fc9025cd
#1125
#1144
#1129

**Comment by nickbabcock** [comment]
(Thinking out loud here). I think the best approach may be a `registerDefaultExceptionMappers` in the `ResourceTestRuleBuilder`, which by default is enabled, and will register all the default exception mappers like the `registerDefaultExceptionMappers` yaml setting (which is also true by default).

How does that sound?

**Comment by lingz** [comment]
That sounds like a fine solution if you want resource test to work out of
the box. Would it be possible to check if exception mappers are already
registered and to only register them if nothing of that type is already
registered?

Bests,
Ling

**Comment by nickbabcock** [comment]
Not to my knowledge :cry: 

```java
ResourceConfig config = /*...*/;
config.register((ExceptionMapper<JerseyViolationException>) exception -> Response.ok().build());
Class<ExceptionMapper<JerseyViolationException>> clazz = (Class) ExceptionMapper.class;
config.isRegistered(clazz); /* returns false! */
```

**Comment by nickbabcock** [comment]
Closed by #1387.

Call `setRegisterDefaultExceptionMappers(false)` when you're building the a TestResourceRule
